File: task1.php
<?php
require_once('data2.php');

function getTeams($list){
    $teamList = [];
    foreach (range(0,count($list)-1) as $p) {
        $item =$list[$p];
        if ($item->isTeam){
            array_push($teamList,$item);

        }
    }
    return$teamList;
}


function exercise1($person, $people){
    $teams = [];
  $teamList = getTeams($people);
  foreach (range(0,count($teamList)-1) as $index){
      $personName = $person->displayName;
      $membersList = $teamList[$index]->members;
      foreach (range(0,count($membersList)-1) as $j){

          if ($membersList[$j]->displayName ==$personName){

              array_push($teams,$teamList[$index]);
          }
      }

  }
  return $teams;

}
$teams = [];
$result = exercise1($alice, $people);
foreach($result as $res) {
    array_push($teams, $res->displayName);
}

echo (json_encode($teams));

File: python_version/task1.py
import data1 

def getTeams(list):

  teamsList = []
  
  for p in range(len(list)):
    item = list[p]
    if item.is_team:
      teamsList.append(item)
      pass
  return teamsList
  

def exercise1(person, people):
  teams = []
  teamList = getTeams(people)
  for index in range(len(teamList)):
    personName = person.displayname
    membersList = teamList[index].members
    for j in range(len(membersList)):
      print(membersList[j].__dict__)
      if membersList[j].displayname == personName:
        teams.append(teamList[index])
      
      
  return teams

print([t.displayname for t in exercise1(data1.alice, data1.people)])
